build/linter/constructor/analyzer.go

```diff
@@ -82,7 +90,7 @@ func getConstructorList(t types.Type) []string {
 		}
 
 		if fieldStruct, ok := named.Underlying().(*types.Struct); ok {
-			ctors = append(ctors, getConstructorList(fieldStruct)...)
+			ctors = append(ctors, getConstructorList(fieldStruct, nil)...)
```

Comment:
Let me clarify. In this PR you added `ignoreFields` to `getConstructorList` to skip some fields when recursively process `struct`. And I think this line is the cause of recursively processing `struct`. We don't need this line because `WithStack` still access the member of `struct`. Can we delete this line and delete the `ignoreFields` logic of `getConstructorList`?

Reply:
`getConstructorList` should be called recursively to avoid implicitly constructing the embedded structs:

```go
type A struct { _ constructor.Constructor `ctor:"NewA"`}
type B struct {A}
type C struct {B}

var c C
```

The `var c C` is not allowed, because it will implicitly create a `A` outside of its constructor. The `WithStack` can do nothing for it as it doesn't appear in the AST.

The `ignoreFields` is added in this PR to allow this situation:

```go
func XFunc() {
    b := B {
        A: NewA()
    }
}
```

Preivously, the `getConstructorList()` of `B` will return `NewA`, and linter will give an error because it's constructed in `XFunc`. However, it should be allowed as the `A` is actually not constructed in `XFunc`, but in `NewA`.